### PR TITLE
#2129 Set proper profile when getting client info

### LINKED ISSUE #2129: Can't get URL for a service deployed to non-default minikube profile
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

Please provide the following details:

**Environment**:

**Minikube version**: v0.22.3
- **OS**: Mac OS X
- **VM Driver**: VirtualBox
- **ISO version**: minikube-v0.23.5.iso

**What happened**:

1. created a custom minikube profile and deployed a service to the minikube cluster (with service type == NodePort).
2. tried to get url of the service using `minikube -p profile-name service --url service-name` command and got an error that server is not reachable (while `minikube -p profile-name status` shows that the server is running).
3. The error message contained a message that minikube failed to reach ip address of the default minikube, while I was trying to get url of the service deployed to the custom minikube profile.

Cursory look into https://github.com/kubernetes/minikube/blob/master/pkg/minikube/service/service.go#L69-L70 shows that minikube does not handle custom profiles correctly, it always sets the kubernetes profile name to 'minikube', instead of the one allocated to the minikube profile.

**What you expected to happen**:
Get url of the service

**How to reproduce it** (as minimally and precisely as possible):
see above

**Output of `minikube logs` (if applicable)**:

**Anything else do we need to know**:

### PR DESCRIPTION
**Pull request by ulyssessouza**
Fixes #2129 

- Set proper profile (instead of default 'minikube') when getting client info

### CODE REVIEW

**Comment by r2d4** [comment]
@minikube-bot ok to test

File: pkg/minikube/service/service.go [hunk 1]
```diff
@@ -64,18 +66,19 @@ func (k *K8sClientGetter) GetCoreClient() (corev1.CoreV1Interface, error) {
 
 func (*K8sClientGetter) GetClientset() (*kubernetes.Clientset, error) {
 	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	profile := viper.GetString(config.MachineProfile)
 	configOverrides := &clientcmd.ConfigOverrides{
 		Context: clientcmdapi.Context{
-			Cluster:  "minikube",
+			Cluster:  profile,
 			AuthInfo: "minikube",
```
Comment (patrikerdes) [inline pkg/minikube/service/service.go]: `AuthInfo` also uses the profile name, so `AuthInfo` should also be set to `profile` on line 70, see the function `PopulateKubeConfig` https://github.com/kubernetes/minikube/blob/master/pkg/util/kubeconfig/config.go#L77-L82.

The following is the result when `AuthInfo` is set to `minikube`:
```
$ out/minikube config set bootstrapper kubeadm
$ out/minikube profile cluster2
minikube profile was successfully set to cluster2
$ out/minikube start
Starting local Kubernetes v1.8.0 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Downloading kubeadm v1.8.0
Downloading kubelet v1.8.0
Finished Downloading kubelet v1.8.0
Finished Downloading kubeadm v1.8.0
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
E1120 08:51:44.524528    5936 start.go:265] Error starting cluster:  timed out waiting to unmark master: getting core client: getting clientset: Error creating kubeConfig: auth info "minikube" does not exist

```

**Comment by ulyssessouza** [comment]
Hello, the build is failing on OSX (in some cases) and Windows. Can you just help me checking if it's something related to my code? 
I don't have an OSX or a Windows to test it :(

And anyways, my code doesn't look like doing anything OS specific... So maybe just an intermittent build problem... I don't know...

**Comment by gbraad** [comment]
@r2d4 Seems issues on the OSX node:
```
		(minikube) DBG | virtio_net: Could not create vmnet interface, permission denied or no entitlement?
		E1120 00:47:53.405394   67418 start.go:150] Error starting host: Error creating host: Error creating machine: Error in driver during machine creation: vmnet: error from vmnet.framework: -1
```

Also on the virtualbox on Windows host:
```
Errors occurred deleting machine:  Error deleting host: minikube: Error loading host from store: open C:\Users\jenkins\.minikube\machines\minikube\config.json: The system cannot find the file specified.
Remove-Item : Cannot remove item C:\Users\jenkins\.minikube\machines\minikube\minikube\Logs\VBox.log: The process 
```

Maybe we should add a delete action of the folder in the build script. We had the same issue for Minishift that vbox.log was still there, which causes libmachine to think the docker machine was actually still configured (hence the error of the missing `config.json`).

@ulyssessouza these errors are not related to your code as far as I can see.

**Comment by r2d4** [comment]
Thanks!

**Pull request merged**